packages/sdk-demo/src/components/Layout/ThemeSelector.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useRef, useState } from 'react';
+
+import { useThemeContext } from '@/context/themeContext';
+import type { I18n } from '@lingui/core';
+import { t } from '@lingui/macro';
+import { useLingui } from '@lingui/react';
+import { useRootElements } from '@monite/sdk-react';
+import DarkModeIcon from '@mui/icons-material/DarkMode';
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
+import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
+import LightModeIcon from '@mui/icons-material/LightMode';
+import {
+  Button,
+  Typography,
+  Menu,
+  MenuItem,
+  Divider,
+  FormControlLabel,
+  Switch,
+} from '@mui/material';
+
+const getButtonLabel = (themeIndex: string, i18n: I18n) => {
+  switch (themeIndex) {
+    case 'material':
+      return t(i18n)`Material UI`;
+    case 'monite':
+      return t(i18n)`Monite`;
+    default:
+      return t(i18n)`Material UI`;
+  }
+};
+
+export const ThemeSelector = () => {
+  const { i18n } = useLingui();
+  const { root } = useRootElements();
+
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const buttonRef = useRef<HTMLButtonElement>(null);
+
+  const { themeIndex, colorMode, setTheme, toggleColorMode } =
```

Comment:
Let's move usage of `setTheme()` on the top of `ThemeSelector`, to make it compatible with Next.js:

```tsx
<ThemeSelector onChange={setTheme} />
```

Reply:
Btw, I changed my mind. We will wrap Next.js App with the same Provider and themes list from `@team-monite/themes`, and reuse the logic.

Reply:
ok